src/Services/Spellcheckers/Aspell.php

```diff
@@ -101,14 +101,45 @@ private function run(string $text): array
 
         $output = $process->getOutput();
 
-        return array_values(array_map(function (string $line): Misspelling {
-            [$wordMetadataAsString, $suggestionsAsString] = explode(':', trim($line));
+        $lines = $this->getAspellLinesFromOutput($output);
+
+        return $this->getMisspellingsFromOutputLines($lines);
+    }
+
+    /**
+     * Gets Aspell output lines as an array
+     *
+     * @return array<int, string>
+     */
+    private function getAspellLinesFromOutput(string $output): array
+    {
+        return array_values(
+            array_filter(
+                explode(PHP_EOL, trim($output)), fn (string $line): bool => ! str_starts_with($line, '@(#)') && $line !== '*'
+            )
+        );
+    }
 
-            $word = explode(' ', $wordMetadataAsString)[1];
-            $suggestions = explode(', ', trim($suggestionsAsString));
+    /**
+     * Maps output lines to an array of Misspellings if suggestions exist, null if no suggestions or an empty array if spelling is correct
+     *
+     * @param  array<int, string>  $lines
+     * @return array <int, Misspelling>
+     */
+    private function getMisspellingsFromOutputLines(array $lines): array
+    {
+        return array_values(array_map(function (string $line): Misspelling {
+            if (preg_match('/#\\s(\\w*)\\sd*/', $line, $matches)) {
+                $word = $matches[1];
+                $suggestions = [];
+            } else {
+                [$wordMetadataAsString, $suggestionsAsString] = explode(':', trim($line));
+                $word = explode(' ', $wordMetadataAsString)[1];
+                $suggestions = explode(', ', trim($suggestionsAsString));
+            }
```

Comment:
If you wanted to, we could turn this if-else block into two `match()` statements.

This way we do not rely on the variables from inside the if block in the return statement and clearly separate the aggregation of the `$word` and the` $suggestions`.

```suggestion
            $word = match (true) {
                preg_match('/#\\s(\\w*)\\sd*/', $line, $matches) => $matches[1],
                default => explode(' ', explode(':', trim($line))[0])[1],
            };
            
            $suggestions = match (true) {
                preg_match('/#\\s(\\w*)\\sd*/', $line, $matches) => [],
                default => explode(', ', trim(explode(':', trim($line))[1])),
            };
```